lib/AppPluginCompose/index.js

```
@@ -115,9 +116,38 @@ class AppPluginCompose extends AppPlugin {
 
 			driverJson.id = driverId;
 
+			function extendSetting(settingsTemplates, rootObj, settingObj, index) {
+				if (settingObj.type === 'group') {
+					for (let childSettingId in settingObj.children) {
+						extendSetting(settingsTemplates, settingObj.children, settingObj.children[childSettingId], childSettingId);
+					}
+				} else if (settingObj.$extends) {
+					let templateId = settingObj.$extends;
+
+					let settingTemplate = {};
+					if (settingsTemplates.hasOwnProperty(templateId)) {
+						settingTemplate = settingsTemplates[templateId];
+					}
```

Maybe we want to throw an error when the extends template does not exist (this does happen for driver templates)

---

Fixed